common/lib/xmodule/xmodule/x_module.py

```diff
@@ -398,7 +398,7 @@ def get_content_titles(self):
         the display_names of all its leaf nodes.
         """
         if self.has_children:
-            return sum((child.get_content_titles() for child in self.get_children()), [])
+            return sum((child.get_content_titles() for child in self.get_display_items()), [])
```

Comment:
On [line](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/views.py#L467)  `FieldDataCache.cache_for_descriptor_descendents` is called. At this point `ModuleSystem` is not setup and in `get_children` `has_access` is not called and the children's are cached on itself.

Now after `ModuleSystem` has been setup, `get_children` will return its cached list instead of calling `get_block` again which would in turn call `get_module`